### DISCUSSION TITLE
[Issue]: 6800XT not being detected

### DISCUSSION THREAD

**Discussion by boarrr** [vladmandic/sdnext#2582] [Q&A]
### Issue Description

I'm trying to install SD on my PC for image generation, I got it working on Windows using DirectML, however it has a lot of issues with VRAM usage and poor performance. So I decided to dual boot a Linux system in order to install.

But for some reason no matter what I do to try and install it on Linux, my GPU is not being detected by pytorch. For reference I have a Ryzen 5800X3D processor, and an RX 6800 XT GPU. When I try to install the AMD drivers other than the ones default to the Ubuntu environment, I kept running into a blackscreen issue. 

Running any form of webui, such as Automatic1111 or SD Next returns an error where torch can't find my GPU and is asking me to skip the CUDA checks. If I put in the command to skip the cuda checks, it will get past that check however it well then throw a warning regarding "No HIP", and it will default to using my CPU instead of GPU.

My suspicion is that it's to do with the drivers I'm installing, but I've tried multiple different ways to install it. If I use ROCMinfo it shows my GPU accurately, but if I do Radeontop to see my usage it doesn't detect my GPU name and says unknown, but it still shows the stats correctly such as VRAM and pipeline.

I have tried all the standard recommended solutions, such as forcing HSA override etc. If anyone has experience with this card and Linux and can help me I'd greatly appreciate it. I know it is possible as I can see benchmarks of my card on vladmandics benchmarks hitting up to 10 it/s.

### Version Platform Description

Ubuntu 22.04, Ubuntu 20.04, Pop OS 22.04. 

Google Chrome, Firefox

Rocm 5.1 - 5.7 all tried.

### Backend

Original

### Branch

Master

### Model

SD 1.5

### Acknowledgements

- [X] I have read the above and searched for existing issues
- [X] I confirm that this is classified correctly and its not an extension issue

**Comment by onlyreportingissues** [comment]
HIP is kinda like a layer that makes CUDA code readable / usable with ROCm (AMD). I imagine it somewhat like Windows program + Wine -> Windows program usable on Linux.

I'm not using Ubuntu, but this guide looks like it got you covered: https://www.reddit.com/r/ROCm/comments/177pwxv/comment/k4um67g/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

The official guide on AMD for Ubuntu suggests installing amdgpu dkms, which is not necessary.

Edit: 
I noticed that he left out `export HSA_OVERRIDE_GFX_VERSION=10.3.0` command for RDNA2 / RX 6000 series.
I recommend adding it to the bottom of your `.bashrc` file in your home folder, if your GPU isn't officially supported. Make invisible files visible in your file manager settings then you will see it. 

It's enabling ROCm support for your graphics card if your graphics card can handle it, even if your graphics card may be officially unsupported.

**Reply by tornado73** [reply]
+
when ROCM is integrated into the system core, the lasing performance drops by an average of 8.5 percent

**--dkms**
torch:2.2.0.dev20231128+rocm5.7 Autocast half xformers: diffusers:0.23.1 transformers:4.35.2

512х512
100%|███████████████████████████████████████████| 30/30 [00:05<00:00, 5.06it/s]

512х768
100%|███████████████████████████████████████████| 30/30 [00:09<00:00, 3.07it/s]

768x768
100%|███████████████████████████████████████████| 30/30 [00:20<00:00, 1.49it/s]

768x1024
100%|███████████████████████████████████████████| 30/30 [00:48<00:00, 1.63s/it]

1024x1024
100%|███████████████████████████████████████████| 11/11 [00:28<00:00, 2.63s/it]


**--no-dkms**
torch:2.2.0.dev20231128+rocm5.7 Autocast half xformers: diffusers:0.23.1 transformers:4.35.2

512x512
100%|███████████████████████████████████████████| 30/30 [00:05<00:00, 5.30it/s] ---5.5%

768x768
100%|███████████████████████████████████████████| 30/30 [00:09<00:00, 3.15it/s] ---4.6%

768x1024
100%|███████████████████████████████████████████| 30/30 [00:42<00:00, 1.40s/it] ---13,5%

1024x1024
100%|███████████████████████████████████████████| 30/30 [01:11<00:00, 2.38s/it] ---10%

sudo usermod -a -G render <username> 
sudo usermod -a -G video <username>       


**install script**

sudo apt update
wget https://repo.radeon.com/amdgpu-install/5.7.1/ubuntu/jammy/amdgpu-install_5.7.50701-1_all.deb

sudo apt install ./amdgpu-install_5.7.50701-1_all.deb
sudo amdgpu-install --usecase=rocm,hip,mllib --no-dkms

nano ~/.bashrc 
![28021120](https://github.com/vladmandic/automatic/assets/58679489/21ae393f-8827-4eab-8024-8e8b4ff53b0b)

**Comment by vladmandic** [comment]
> When I try to install the AMD drivers other than the ones default to the Ubuntu environment, I kept running into a blackscreen issue.

you need to sort out to get official amd drivers and rocm installed. i cannot help with that.

**Reply by onlyreportingissues** [reply]
The official docs are a little confusing. The proprietary graphics driver isn't needed (see screenshot), what's important is to get the ROCm & HIP files that are required.
![Bildschirmfoto vom 2023-12-05 20-21-31](https://github.com/vladmandic/automatic/assets/103491514/59612b88-b90b-40af-adee-1ecdc2490cae)

**Comment by tornado73** [comment]
> I know it is possible as I can see benchmarks of my card on vladmandics benchmarks hitting up to 10 it/s.

It looks like science fiction, on the RX 6800 the maximum that I squeezed out on the standard automatic1111, in April on Ubuntu 20.04 + ROCM 5.4.2 -7.2it/s


Now is the maximum
100%|███████████████████████████████████████████| 30/30 [00:05<00:00, 5.30it/s]
-((

Tomorrow I'll rearrange the system and play with the old drivers and describe the result

**Comment by tornado73** [comment]
Ubuntu 20.04.6 LTS
torch:2.0.1+rocm5.4.2 Autocast half xformers: diffusers:0.24.0 transformers:4.35.2
device:AMD Radeon RX 6800 (1) hip:5.4.22803-474e8620 16GB
realisticVisionV20
100%|███████████████████████████████████████████| 30/30 [00:04<00:00,  6.04it/s]

vs 

Ubuntu 22.04.3 LTS
orch:2.2.0.dev20231128+rocm5.7 Autocast half xformers: diffusers:0.23.1 transformers:4.35.2
realisticVisionV20
100%|███████████████████████████████████████████| 30/30 [00:05<00:00, 5.30it/s]

-12% new rocm ,new torch,new libraries...very good -(((
